csharp/src/Ice/Current.cs

```diff
@@ -15,13 +15,15 @@ public class Current
         public Connection? Connection { get; }
         // TODO: should this be a IReadOnlyDictionary<string, string>?
         public Dictionary<string, string> Context { get; }
-        public Encoding Encoding { get; }
-        public string Facet { get; }
-        public Identity Identity { get; }
-        public bool IsIdempotent { get; }
+        public Encoding Encoding => Request.Encoding;
+        public string Facet => Request.Facet;
+        public Identity Identity => Request.Identity;
+        public bool IsIdempotent => Request.IsIdempotent;
         public bool IsOneway { get; }
-        public string Operation { get; }
-        public Protocol Protocol { get; }
+        public string Operation => Request.Operation;
+        public Protocol Protocol => Request.Protocol;
+
+        internal IncomingRequestFrame Request { get; }
```

Comment:
It would call this property `IncomingRequestFrame`. I find `Request` ambiguous. 
And should it be public? If we make it public, we could remove rarely used Current properties such as `Encoding`, `IsIdempotent` and maybe `Protocol`.

Reply:
Given that Current is a server side construct, there was no ambiguity to me... but I renamed Request to IncomingRequestFrame for clarity. I kept it protected for now, we could indeed decide to make it public and further simplify Current.